opencv.yaml

```diff
@@ -0,0 +1,136 @@
+    items:
+      calib3d: Camera calibration and 3D reconstruction.
+      core: Essential structures and functions.
+      dnn: Deep neural network inference.
+      features2d: Keypoint detection and description.
+      flann: Fast nearest neighbor searches.
+      highgui: Basic GUI and image display.
+      imgcodecs: Image file read/write.
+      imgproc: Core image processing routines.
+      ml: Traditional machine learning algorithms.
+      objdetect: Object detection methods.
+      photo: Image enhancement and denoising.
+      stitching: Panorama image stitching.
+      video: Video analysis and processing.
+      videoio: Video capture and output.
+
+environment:
+  environment:
+    CFLAGS: "$CFLAGS -g1"
+    CXXFLAGS: "$CXXFLAGS -g1"
+    CC: clang
+    CXX: clang++
+  contents:
+    packages:
+      - binutils
+      - build-base
+      - busybox
+      - ca-certificates-bundle
+      - clang
+      - cmake
+      - eigen-dev
+      - glibc
+      - harfbuzz-dev
+      - hdf5-dev
+      - libtbb-dev
+      - lld
+      - ninja-build
+      - openblas-dev
+      - opencl-dev
+      - samurai
+
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/opencv/opencv
+      expected-commit: 31b0eeea0b44b370fd0712312df4214d4ae1b158
+      tag: ${{package.version}}
+
+  - uses: cmake/configure
+    name: Cmake Configure
+    with:
+      output-dir: build
+      opts: |
+        -DENABLE_BUILD_HARDENING=ON \
+        -DENABLE_THIN_LTO=ON \
+        -DWITH_OPENCL=ON \
+        -DWITH_OPENGL=ON \
+        -DWITH_TBB=ON \
+        -DWITH_VULKAN=ON \
+        -DWITH_QT=ON \
+        -DWITH_ADE=OFF \
+        -DWITH_opencv_gapi=OFF \
+        -DWITH_IPP=OFF \
+        -DBUILD_WITH_DEBUG_INFO=ON \
+        -DBUILD_TESTS=OFF \
+        -DBUILD_PERF_TESTS=OFF \
+        -DBUILD_EXAMPLES=OFF \
+        -DINSTALL_C_EXAMPLES=OFF \
+        -DINSTALL_PYTHON_EXAMPLES=OFF \
+        -DOPENCV_SKIP_PYTHON_LOADER=ON \
+        -DOPENCV_GENERATE_PKGCONFIG=ON \
+        -DOPENCV_ENABLE_NONFREE=OFF \
+        -DOPENCV_GENERATE_SETUPVARS=OFF \
+        -DEIGEN_INCLUDE_PATH=/usr/include/eigen3 \
+        -DCMAKE_SKIP_INSTALL_RPATH=ON
+
+  - runs: |
+      git submodule update --init --recursive
+
+  - uses: cmake/build
+    name: Building opencv
+    with:
+      output-dir: build
+
+  - runs: |
+      ctest --test-dir build --output-on-failure
+
+  - uses: cmake/install
+    name: Install opencv
+    with:
+      output-dir: build
+
+  - uses: strip
+
+subpackages:
+  - name: ${{package.name}}-dev
+    description: "Open source computer vision and machine learning library (development files)"
+    pipeline:
+      - uses: split/dev
+
+  - range: libs
+    name: libopencv-${{range.key}}
+    description: opencv's ${{range.value}} library
+    pipeline:
+      - runs: |
+          mkdir -p ${{targets.subpkgdir}}/usr/lib
+          mv ${{targets.destdir}}/usr/lib/libopencv_${{range.key}}.so.* ${{targets.subpkgdir}}/usr/lib
```

Comment:
Can we have some of sub package based test to ensure that these libs exist in the expected directory, if it’s too complex due to range thing, then we can discuss alternatives but it will be good to have it